lib/service/service.go

```diff
@@ -2605,6 +2605,43 @@ func (process *TeleportProcess) initAuthService() error {
 		}
 	}
 
+	// We mark the process state as starting until the auth backend is ready.
+	// If cache is enabled, this will wait for the cache to be populated.
+	// We don't want auth to say its ready until its cache is populated,
+	// else a rollout might progress too quickly and cause backend pressure
+	// and outages.
+	{
+		component := teleport.Component(teleport.ComponentAuth, "backend")
+		process.ExpectService(component)
+		process.RegisterFunc("auth.wait-for-backend", func() error {
+			start := process.Clock.Now()
+
+			w, err := authServer.NewWatcher(process.ExitContext(), types.Watch{
```

Comment:
I think we must have a retry loop around this, or we might end up with a process that's forever unready even if things are resolved later on.

Reply:
done